Makefile

```diff
@@ -79,115 +84,51 @@ test:
-	git reset HEAD ${SOURCE_DIR}/resources/smcp-templates/v2.1
-
-.PHONY: collect-2.1-charts
-collect-2.1-charts:
-	mkdir -p ${HELM_OUT_DIR}
-	cp -rf ${RESOURCES_DIR}/helm/v2.1 ${HELM_OUT_DIR}
-
-.PHONY: collect-2.1-templates
-collect-2.1-templates:
-	mkdir -p ${TEMPLATES_OUT_DIR}/v2.1
-	cp ${RESOURCES_DIR}/smcp-templates/v2.1/${BUILD_TYPE} ${TEMPLATES_OUT_DIR}/v2.1/default
-	find ${RESOURCES_DIR}/smcp-templates/v2.1/ -maxdepth 1 -type f ! -name "maistra" ! -name "servicemesh" |xargs cp -t ${TEMPLATES_OUT_DIR}/v2.1
-
-################################################################################
-# maistra v2.2
-################################################################################
-.PHONY: update-remote-maistra-2.2
-update-remote-maistra-2.2:
-ifeq "${OFFLINE_BUILD}" "false"
-	git remote set-branches --add ${GIT_UPSTREAM_REMOTE} maistra-2.2
-	git fetch ${GIT_UPSTREAM_REMOTE} maistra-2.2:maistra-2.2
-endif
-
-.PHONY: update-2.2-charts
-update-2.2-charts: update-remote-maistra-2.2
-	git checkout ${GIT_UPSTREAM_REMOTE}/maistra-2.2 -- ${SOURCE_DIR}/resources/helm/v2.2
-	git reset HEAD ${SOURCE_DIR}/resources/helm/v2.2
-
-.PHONY: update-2.2-templates
-update-2.2-templates: update-remote-maistra-2.2
-	git checkout ${GIT_UPSTREAM_REMOTE}/maistra-2.2 -- ${SOURCE_DIR}/resources/smcp-templates/v2.2
-	git reset HEAD ${SOURCE_DIR}/resources/smcp-templates/v2.2
-
-.PHONY: collect-2.2-charts
-collect-2.2-charts:
-	mkdir -p ${HELM_OUT_DIR}
-	cp -rf ${RESOURCES_DIR}/helm/v2.2 ${HELM_OUT_DIR}
-
-.PHONY: collect-2.2-templates
-collect-2.2-templates:
-	mkdir -p ${TEMPLATES_OUT_DIR}/v2.2
-	cp ${RESOURCES_DIR}/smcp-templates/v2.2/${BUILD_TYPE} ${TEMPLATES_OUT_DIR}/v2.2/default
-	find ${RESOURCES_DIR}/smcp-templates/v2.2/ -maxdepth 1 -type f ! -name "maistra" ! -name "servicemesh" |xargs cp -t ${TEMPLATES_OUT_DIR}/v2.2
-
-################################################################################
-# maistra v2.3
-################################################################################
-.PHONY: update-2.3-charts
-update-2.3-charts:
-	HELM_DIR=${RESOURCES_DIR}/helm/v2.3 ISTIO_VERSION=1.14.3 ${SOURCE_DIR}/build/download-charts.sh
-
-.PHONY: collect-2.3-charts
-collect-2.3-charts:
+.PHONY: update-charts-%
+update-charts-%:
+	$(eval version:=$(subst update-charts-,,$@))
+	# If we are calling make against current version - download charts.
+	# Otherwise sync from previous branches and explicitly call dependent target with extracted version
+	@ if [[ ${MAISTRA_VERSION} == ${version}* ]]; \
+	then \
+		echo "test"; \
+		HELM_DIR=${RESOURCES_DIR}/helm/v${version} ISTIO_VERSION=${ISTIO_VERSION} ${SOURCE_DIR}/build/download-charts.sh; \
+	else \
+		$(MAKE) -f $(THIS_FILE) update-remote-maistra-${version}; \
+		git checkout ${GIT_UPSTREAM_REMOTE}/maistra-${version} -- ${SOURCE_DIR}/resources/helm/v${version}; \
+		git reset HEAD ${SOURCE_DIR}/resources/helm/v${version}; \
```

Comment:
do you need `--hard`?

Reply:
Didn't think about the code - it was just there :) Do you think it would help us avoid some problems?

Reply:
nah, nevermind